### PR TITLE
SL-19575 Create emoji gallery access icon

### PR DESCRIPTION
**Pull request by LLGuru**

### CODE REVIEW

File: indra/newview/llfloateremojipicker.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+#include "llviewerprecompiledheaders.h"
+
+#include "llfloateremojipicker.h"
+
+#include "llfloaterreg.h"
+#include "llscrolllistctrl.h"
+#include "llscrolllistitem.h"
+#include "llemojidictionary.h"
+
+#pragma optimize ("", off)
```
Comment (callumlinden) [inline indra/newview/llfloateremojipicker.cpp:35]: I expect you have a note about this but don't forget to remove this directive before release.

**Review by callumlinden — APPROVED** [review]
Excellent work Guru and I love how you remembered all the minor details too - e.g. adding the source files to CMake scripts in the alphabetical order vs. at the end.

File: indra/newview/llfloateremojipicker.cpp (hunk 1 above)
Comment (LLGuru) [reply indra/newview/llfloateremojipicker.cpp:35]: Oh, my god! Yes, this artifact should be removed :-)

**Comment by LLGuru** [comment]
> Excellent work Guru and I love how you remembered all the minor details too - e.g. adding the source files to CMake scripts in the alphabetical order vs. at the end.

Thanks

> Don't forget to remove the pragma to turn off optimization.

Already fixed

> I think your editor is reformatting unchanged code (tabs->spaces or vice-versa) - see if you can turn that behavior off.

The project source contains a mix of tabs and spaces in indents
I try to follow the standard used in the neighbouring lines

> Really impressive work.

I'm impressed myself :-D

File: indra/newview/skins/default/xui/en/floater_emoji_picker.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<floater
+  name="emojipicker"
+  title="CHOOSE EMOJI"
+  help_topic="emojipicker"
+  positioning="cascading"
+  legacy_header_height="10"
+  can_resize="false"
+  layout="topleft"
+  height="419"
+  width="200">
```
Comment (akleshchev) [inline indra/newview/skins/default/xui/en/floater_emoji_picker.xml:11]: I think it should be declared as a single instance here, with reusability.

**Pull request merged**